hal/network/ncp/wifi/wlan_hal.cpp

```diff
@@ -182,7 +182,71 @@ int wlan_set_credentials(WLanCredentials* halCred) {
     conf.credentials(std::move(cred));
     const auto mgr = wifiNetworkManager();
     CHECK_TRUE(mgr, SYSTEM_ERROR_UNKNOWN);
-    CHECK(mgr->setNetworkConfig(std::move(conf)));
+    if (!(halCred->flags & WLAN_SET_CREDENTIALS_FLAGS_VALIDATE)) {
+        CHECK(mgr->setNetworkConfig(std::move(conf)));
+        return  0;
+    }
+    // Note: It's maily a copy of joinNewNetwork() in wifi_new.cpp
```

Comment:
Let's have this functionality in a common function then shared by both wlan_hal and `wifi_new.cpp` ?